src/gpu/nvidia/cudnn_reorder_lt_impl.hpp

```diff
@@ -0,0 +1,244 @@
+        cudaStream_t streamId;
+        auto lt_handle = (cublasLtHandle_t)(cublas_handle);
+        CUBLAS_EXECUTE_FUNC(cublasGetStream, cublas_handle, &streamId);
+        int alpha = 1;
+        if (src_scale) {
+            float host_src_scale = 1.0f;
+            CUDA_EXECUTE_FUNC(cuMemcpyAsync, (CUdeviceptr)&host_src_scale,
+                    (CUdeviceptr)src_scale, sizeof(float), streamId);
+            alpha *= host_src_scale;
+        }
+        int beta = beta_;
+        if (dst_scale) {
+            float host_dst_scale = 1.0f;
+            CUDA_EXECUTE_FUNC(cuMemcpyAsync, (CUdeviceptr)&host_dst_scale,
+                    (CUdeviceptr)dst_scale, sizeof(float), streamId);
+            alpha /= host_dst_scale;
+            beta /= host_dst_scale;
+        }
+        cublasOperation_t transform_trans = trans ? CUBLAS_OP_T : CUBLAS_OP_N;
+        CUBLAS_EXECUTE_FUNC(cublasLtMatrixTransformDescSetAttribute,
+                trans_desc_, CUBLASLT_MATRIX_TRANSFORM_DESC_TRANSA,
+                &transform_trans, sizeof(transform_trans));
+        CUBLAS_EXECUTE_FUNC(cublasLtMatrixTransform, lt_handle, trans_desc_,
+                &alpha, src, src_layout_, &beta, dst, dst_layout_, dst,
+                dst_layout_, streamId);
+    }
+
+    ~cublaslt_reorder_t() { cleanup(); }
+
+    void cleanup() {
+        if (src_layout_) {
+            CUBLAS_EXECUTE_FUNC(cublasLtMatrixLayoutDestroy, src_layout_);
+            src_layout_ = nullptr;
+        }
+        if (dst_layout_) {
+            CUBLAS_EXECUTE_FUNC(cublasLtMatrixLayoutDestroy, dst_layout_);
+            dst_layout_ = nullptr;
+        }
+        if (trans_desc_) {
+            CUBLAS_EXECUTE_FUNC(
+                    cublasLtMatrixTransformDescDestroy, trans_desc_);
+            trans_desc_ = nullptr;
+        }
+    }
+
+    void init_scratchpad(reorder_pd_t *pd,
+            const impl::primitive_desc_t *generic_reorder_desc) {
+
+        auto scratchpad = pd->scratchpad_registry().registrar();
+        scratchpad.book(memory_tracking::names::key_nested,
+                generic_reorder_desc->scratchpad_registry());
+        if (src_scratch_size_) {
+            scratchpad.book(
+                    memory_tracking::names::key_reorder_cublaslt_src_float,
+                    src_scratch_size_, 1, 256);
+        }
+        if (dst_scratch_size_) {
+            scratchpad.book(
+                    memory_tracking::names::key_reorder_cublaslt_dst_float,
+                    dst_scratch_size_, 1, 256);
+        }
+    }
+
+protected:
+    cudaDataType_t src_data_type_;
+    cudaDataType_t dst_data_type_;
+    int dims_[DNNL_MAX_NDIMS];
+    float beta_ = 0.0f;
+
+    cublasLtMatrixTransformDesc_t trans_desc_;
+    cublasLtMatrixLayout_t src_layout_;
+    cublasLtMatrixLayout_t dst_layout_;
+    uint64_t src_scratch_size_ = 0;
+    uint64_t dst_scratch_size_ = 0;
+
+    int batch_count_;
+    uint64_t row_, col_;
+
+    cublasLtOrder_t ampere_order_ = CUBLASLT_ORDER_COL32_2R_4R4;
```

Comment:
in the documentation of clublasLt, it seems that 2 different layouts are possible for the B matrix on ampere:
```
CUBLASLT_ORDER_COL4_4R2_8C (on Turing or Ampere architecture) or CUBLASLT_ORDER_COL32_2R_4R4 (on Ampere architecture) for matrix B
```
Any reason to limit to a single of these two layouts for ampere?